packages/NodeTypeResolver/src/TypeContext.php

```diff
@@ -78,13 +78,12 @@ public function addVariableWithTypes(string $variableName, array $variableTypes)
         $this->variableTypes[$variableName] = $variableTypes;
     }
 
-    public function enterClassLike(ClassLike $classLikeNode): void
+    public function enterClassLike(Class_ $classLikeNode): void
```

Comment:
This should also accept interface and traits as well. Maybe we could add test case to cover it?

Reply:
Gonna add this tests than

Reply:
I'll rollback this one for now and fix it in a different PR with tests :)